src/test/java/com/app/parallel/android/BaseTest.java

```diff
@@ -0,0 +1,98 @@
+package com.app.parallel.android;
+
+import com.app.DeviceDetails;
+import io.appium.java_client.MobileDriver;
+import io.appium.java_client.MobileElement;
+import io.appium.java_client.android.AndroidDriver;
+import io.restassured.authentication.PreemptiveBasicAuthScheme;
+import io.restassured.builder.RequestSpecBuilder;
+import io.restassured.builder.ResponseSpecBuilder;
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.remote.DesiredCapabilities;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.BeforeSuite;
+
+import java.lang.reflect.Method;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.concurrent.ThreadLocalRandom;
+
+import static io.restassured.RestAssured.*;
+import static java.util.stream.Collectors.toList;
+
+public class BaseTest {
+
+    private static final ThreadLocal<MobileDriver<MobileElement>> driverThread = new ThreadLocal<>();
+
+    private static final String USERNAME = System.getenv("BROWSERSTACK_USERNAME");
+    private static final String ACCESS_KEY = System.getenv("BROWSERSTACK_ACCESS_KEY");
+    private static final String URL = "http://hub-cloud.browserstack.com/wd/hub";
+    private static final LocalDateTime TIMESTAMP = LocalDateTime.now();
+
+    public static MobileDriver<MobileElement> getMobileDriver() {
+        return driverThread.get();
+    }
+
+    @BeforeSuite(alwaysRun = true)
+    public void setupApp() {
```

Comment:
A common function for android/ios app upload would be better

Reply:
Added AppUtils class which has uploadApp method to upload specific apps. Used this method in the setupApp method.